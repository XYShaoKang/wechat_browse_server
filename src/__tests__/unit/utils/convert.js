import {
  createWeChatUserCreateInput,
  pickUserName,
  createChatRoomCreateInput,
  createMessageInput,
} from '../../../utils'

const userData = [
  {
    username: 'aaaa',
    alias: '',
    conRemark: '',
    nickname: 'AAAA',
    avatar: {
      thumbnailImg: 'http://a.com/0',
      bigImg: 'http://a.com/1',
    },
  },
  {
    username: 'bbbb',
    alias: '',
    conRemark: '',
    nickname: 'BBBB',
    avatar: {
      thumbnailImg: 'http://b.com/0',
      bigImg: 'http://b.com/1',
    },
  },
]
const chatRoomData = [
  {
    username: 'abc@chatroom',
    nickname: 'ABC',
    displayName: 'AAAA縲。BBB縲，CCC',
    owner: 'aaaa',
    memberList: ['aaaa', 'bbbb', 'cccc'],
    modifyTime: '2018-10-13T13:10:31.025Z',
    avatar: null,
  },
  {
    username: 'def@chatroom',
    nickname: 'DEF',
    displayName: 'DDDD縲・EEE縲：FFF',
    owner: 'dddd',
    memberList: ['dddd', 'eeee', 'ffff'],
    modifyTime: '2018-10-13T13:10:31.025Z',
    avatar: {
      thumbnailImg: 'http://def.com/0',
      bigImg: 'http://def.com/1',
    },
  },
]
const weChatUsers = [
  { username: 'aaaa', id: '0' },
  { username: 'bbbb', id: '1' },
  { username: 'cccc', id: '2' },
  { username: 'dddd', id: '3' },
  { username: 'eeee', id: '4' },
  { username: 'ffff', id: '5' },
]
const messageData = [
  {
    msgSvrId: '1000000000000000001',
    isSend: 0,
    type: 1,
    talker: 'aaaa',
    createTime: '2016-05-19T00:00:22.935Z',
  },
  {
    msgSvrId: '1000000000000000002',
    isSend: 1,
    type: 1,
    talker: 'bbbb',
    createTime: '2016-05-19T00:00:22.935Z',
    chatRoom: 'bbbb@chatroom',
  },
  {
    msgSvrId: '1000000000000000003',
    isSend: 0,
    type: 1,
    talker: 'bbbb@chatroom',
    createTime: '2016-07-10T11:40:28.809Z',
    chatRoom: 'bbbb@chatroom',
  },
]
describe('convert', () => {
  it('createWeChatUserCreateInput', () => {
    const chatUserCreateInput = createWeChatUserCreateInput(userData)

    expect(chatUserCreateInput).toMatchSnapshot()
  })
  it('pickUserName', () => {
    /** @type {string[]} */
    const usernames = pickUserName(chatRoomData)

    expect(usernames).toMatchSnapshot()
  })
  it('pickUserName', () => {
    const chatRoomCreateInput = createChatRoomCreateInput(weChatUsers)(
      chatRoomData,
    )

    expect(chatRoomCreateInput).toMatchSnapshot()
  })
  it('createMessageInput', () => {
    const messageInput = createMessageInput(messageData)

    expect(messageInput).toMatchSnapshot()
  })
})
