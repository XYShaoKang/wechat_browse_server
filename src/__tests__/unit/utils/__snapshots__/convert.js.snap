// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convert createMessageInput 1`] = `
Array [
  Object {
    "chatRoom": null,
    "createTime": "2016-05-19T00:00:22.935Z",
    "isSend": 0,
    "msgSvrId": "1000000000000000001",
    "talker": Object {
      "connect": Object {
        "username": "aaaa",
      },
    },
    "type": 1,
  },
  Object {
    "chatRoom": Object {
      "connect": Object {
        "username": "bbbb@chatroom",
      },
    },
    "createTime": "2016-05-19T00:00:22.935Z",
    "isSend": 1,
    "msgSvrId": "1000000000000000002",
    "talker": Object {
      "connect": Object {
        "username": "bbbb",
      },
    },
    "type": 1,
  },
  Object {
    "chatRoom": Object {
      "connect": Object {
        "username": "bbbb@chatroom",
      },
    },
    "createTime": "2016-07-10T11:40:28.809Z",
    "isSend": 0,
    "msgSvrId": "1000000000000000003",
    "talker": null,
    "type": 1,
  },
]
`;

exports[`convert createWeChatUserCreateInput 1`] = `
Array [
  Object {
    "alias": "",
    "avatar": Object {
      "create": Object {
        "bigImg": Object {
          "create": Object {
            "url": "http://a.com/1",
          },
        },
        "thumbnailImg": Object {
          "create": Object {
            "url": "http://a.com/0",
          },
        },
      },
    },
    "conRemark": "",
    "nickname": "AAAA",
    "username": "aaaa",
  },
  Object {
    "alias": "",
    "avatar": Object {
      "create": Object {
        "bigImg": Object {
          "create": Object {
            "url": "http://b.com/1",
          },
        },
        "thumbnailImg": Object {
          "create": Object {
            "url": "http://b.com/0",
          },
        },
      },
    },
    "conRemark": "",
    "nickname": "BBBB",
    "username": "bbbb",
  },
]
`;

exports[`convert pickUserName 1`] = `
Array [
  "aaaa",
  "bbbb",
  "cccc",
  "aaaa",
  "dddd",
  "eeee",
  "ffff",
  "dddd",
]
`;

exports[`convert pickUserName 2`] = `
Array [
  Object {
    "avatar": null,
    "displayName": "AAAA縲。BBB縲，CCC",
    "memberList": Object {
      "connect": Array [
        Object {
          "id": "0",
        },
        Object {
          "id": "1",
        },
        Object {
          "id": "2",
        },
      ],
    },
    "modifyTime": "2018-10-13T13:10:31.025Z",
    "nickname": "ABC",
    "owner": Object {
      "connect": Object {
        "id": "0",
      },
    },
    "username": "abc@chatroom",
  },
  Object {
    "avatar": Object {
      "create": Object {
        "bigImg": Object {
          "create": Object {
            "url": "http://def.com/1",
          },
        },
        "thumbnailImg": Object {
          "create": Object {
            "url": "http://def.com/0",
          },
        },
      },
    },
    "displayName": "DDDD縲・EEE縲：FFF",
    "memberList": Object {
      "connect": Array [
        Object {
          "id": "3",
        },
        Object {
          "id": "4",
        },
        Object {
          "id": "5",
        },
      ],
    },
    "modifyTime": "2018-10-13T13:10:31.025Z",
    "nickname": "DEF",
    "owner": Object {
      "connect": Object {
        "id": "3",
      },
    },
    "username": "def@chatroom",
  },
]
`;
